src/csharp/Microsoft.Spark/Sql/RowCollector.cs

```diff
@@ -30,14 +25,9 @@ public IEnumerable<Row> Collect(ISocketWrapper socket)
             Stream inputStream = socket.InputStream;
 
             int? length;
-            while (((length = SerDe.ReadBytesLength(inputStream)) != null) &&
-                (length.GetValueOrDefault() > 0))
+            while (((length = SerDe.ReadBytesLength(inputStream)) != null) && (length.Value > 0))
             {
-                MaxLengthReadStream readStream = s_slicedReadStream ??
-                    (s_slicedReadStream = new MaxLengthReadStream());
-
-                readStream.Reset(inputStream, length.GetValueOrDefault());
-                var unpickledObjects = PythonSerDe.GetUnpickledObjects(readStream);
+                object[] unpickledObjects = PythonSerDe.GetUnpickledObjects(inputStream, length.Value);
```

Comment:
Why did you change these from GetValueOrDefault to Value?

Reply:
Because the loop checks if it's not null, and when it's not null it is guaranteed to have a value.

```cs
while (((length = SerDe.ReadBytesLength(inputStream)) != null) && (length.Value > 0))
```

https://github.com/dotnet/spark/blob/35048c9cfa324336497055d6543fa46b8d2de247/src/csharp/Microsoft.Spark/Interop/Ipc/SerDe.cs#L215-L230

Reply:
@adamsitnik, have you looked at the implementations of `Nullable<T>.Value` vs `Nullable<T>.GetValueOrDefault`?

Reply:
I did not and I did not know about the difference.  I am going to revert it.

BTW we should really write a Roslyn analyzer for this kind of micro-optimizations.